internal/metadata/icon.go

```diff
@@ -0,0 +1,41 @@
+package metadata
+
+import (
+	"bytes"
+	"image/png"
+	"strconv"
+
+	"github.com/nfnt/resize"
+
+	"fyne.io/fyne/v2"
+)
+
+func ScaleIcon(data fyne.Resource, size int) fyne.Resource {
+	img, err := png.Decode(bytes.NewReader(data.Content()))
+	if err != nil {
+		fyne.LogError("Failed to decode app icon", err)
+		return data
+	}
+
+	if img.Bounds().Dx() <= size {
+		return data
+	}
+
+	sized := resize.Resize(uint(size), uint(size), img, resize.Lanczos3)
```

Comment:
Same concern here, it appears we're resizing to square always rather than maintaining the aspect ratio?

Reply:
Yup, square is the only format we support. Most OS expect square so resizing here may even be an improvement if people accidentally pass off-square...